pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -226,7 +226,12 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
 
     // Calculate instance partitions map
     Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap;
+    Map<InstancePartitionsType, InstancePartitions> existingInstancePartitionsMap = null;
     try {
+      if (reassignInstances) {
+        existingInstancePartitionsMap =
```

Comment:
(Code format) Please reformat the changes with [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)